### PR TITLE
feat(web): Web containers Onboarding

### PR DESCRIPTION
**Pull request by scopsy**
### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/web/config-overrides.js [hunk 1]
```diff
@@ -23,5 +23,35 @@ function overrideConfig(config, env) {
...
+  return {
+    ...config,
+    headers: (req, res, context) => {
+      const defaultHeaders = {
+        'Referrer-Policy': 'no-referrer',
+        'X-XSS-Protection': '1; mode=block',
+        'X-Content-Type-Options': 'nosniff',
+        'Permissions-Policy':
+          'accelerometer=(), camera=(), geolocation=(), gyroscope=(), magnetometer=(), microphone=(), payment=(), usb=(), interest-cohort=()',
+      };
```
Comment (scopsy) [inline apps/web/config-overrides.js:36]: This matches netlify header with our local environment, to better simulat staging and prod env

File: apps/api/src/app/bridge/usecases/preview-step/preview-step.usecase.ts [hunk 2]
```diff
@@ -14,15 +14,16 @@ export class PreviewStep {
 
   async execute(command: PreviewStepCommand) {
     const environment = await this.environmentRepository.findOne({ _id: command.environmentId });
-    if (!environment?.echo.url) {
+    const bridgeUrl = command.bridgeUrl || environment?.echo.url;
```
Comment (scopsy) [inline apps/api/src/app/bridge/usecases/preview-step/preview-step.usecase.ts:17]: This allow passing a dynamic bridge url to the preview function

File: apps/web/netlify.toml [hunk 3]
```diff
@@ -37,11 +37,41 @@
...
+    Referrer-Policy = "no-referrer-when-downgrade"
+    X-Content-Type-Options = "nosniff"
+    Cross-Origin-Opener-Policy = "same-origin"
+    Cross-Origin-Embedder-Policy = "credentialless"
+    Cross-Origin-Resource-Policy = "cross-origin"
+    Permissions-Policy = "accelerometer=(), camera=(), geolocation=(), gyroscope=(), magnetometer=(), microphone=(), payment=(), usb=(), interest-cohort=()"
+    Strict-Transport-Security = '''
+    max-age=63072000;
+    includeSubDomains;
+    preload'''
```
Comment (scopsy) [inline apps/web/netlify.toml:68]: New headers are required for web containers

File: apps/web/package.json [hunk 4]
```diff
@@ -90,6 +93,7 @@
     "date-fns": "^2.29.2",
     "dotenv": "^16.4.5",
     "eslint-plugin-react-hooks": "^4.4.0",
+    "framer-motion": "^11.3.19",
```
Comment (scopsy) [inline apps/web/package.json:96]: Added framer for some motion 🫧

File: apps/web/src/AppRoutes.tsx [hunk 5]
```diff
@@ -53,20 +53,16 @@ import { OrganizationPage } from './pages/settings/organization';
...
 import { WorkflowsStepEditorPageV2 } from './pages/templates/editor_v2/TemplateStepEditorV2';
 import { IS_EE_AUTH_ENABLED } from './config/index';
 import { EnterpriseAuthRoutes } from './ee/clerk/EnterpriseAuthRoutes';
 import { novuOnboardedCookie } from './utils/cookies';
 import { EnterprisePrivatePageLayout } from './ee/clerk/components/EnterprisePrivatePageLayout';
 import { OnboardingPage } from './pages/auth/onboarding/Onboarding';
-import { GetStartedPageV2 } from './studio/components/GetStartedPageV2/index';
+import { PlaygroundPage } from './pages/auth/onboarding/PlaygroundPage';
 import { BillingRoutes } from './pages/BillingPages';
+import { StudioStepEditorPage } from './studio/pages/StudioStepEditorPage';
```
Comment (scopsy) [inline apps/web/src/AppRoutes.tsx:65]: Refactored some of the page names

File: apps/web/src/components/workflow_v2/StepEditorComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+import { IconPlayArrow } from '@novu/design-system';
+import { WorkflowsPanelLayout } from '../../studio/components/workflows/layout/WorkflowsPanelLayout';
+import { WorkflowStepEditorContentPanel } from '../../studio/components/workflows/step-editor/WorkflowStepEditorContentPanel';
+import { WorkflowStepEditorControlsPanel } from '../../studio/components/workflows/step-editor/WorkflowStepEditorControlsPanel';
+import { WORKFLOW_NODE_STEP_ICON_DICTIONARY } from '../../studio/components/workflows/node-view/WorkflowNodes';
+import { BackButton } from '../layout/components/LocalStudioHeader/BackButton';
+import { WorkflowsPageTemplate } from '../../studio/components/workflows/layout/WorkflowsPageTemplate';
+import { OutlineButton } from '../../studio/components/OutlineButton';
+
+export const WorkflowsStepEditor = ({
```
Comment (scopsy) [inline apps/web/src/components/workflow_v2/StepEditorComponent.tsx:7]: A separate stateless component for displaying

File: apps/web/src/pages/get-started/CodeBlock.tsx
Comment (djabarovgeorge) [inline apps/web/src/pages/get-started/CodeBlock.tsx:1]: we need to refactor this one to a simple code block in the near future

File: apps/web/src/pages/get-started/tabs/Setup.tsx [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+
+          return <CodeSnippet command={`npx novu@latest init --secret-key=${(environment as any)?.apiKeys[0].key}`} />;
+        },
+      },
+      {
+        title: 'Run the application server',
+        content: () => {
+          return <CodeSnippet command={`cd my-novu-app && npm run dev`} />;
+        },
+      },
```
Comment (djabarovgeorge) [inline apps/web/src/pages/get-started/tabs/Setup.tsx:36]: i wonder if it is better to merge this step into step 1 `Initialize a new project`

File: apps/web/src/pages/get-started/tabs/Setup.tsx [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+                Once a workflow has been created, we would need to expose it to the <code>serve</code> function so that
+                it will be visible on the Novu Studio.
+              </TextElement>
+              <br /> <br />
+              <CodeEditor
+                height="120px"
+                readonly
+                setCode={() => {}}
+                code={`import { serve } from '@novu/framework/next';
+import { myWorkflow } from '../../novu/workflows';
```
Comment (djabarovgeorge) [inline apps/web/src/pages/get-started/tabs/Setup.tsx:91]: i think we can remove the complexity of telling the user he should import the workflow, all of the code can live in one file for simplicity

File: apps/web/src/pages/get-started/tabs/Setup.tsx [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+    {
+      title: 'Send a test notification',
+      content: () => (
+        <TextElement>
+          Once the Studio is open, navigate to the 'test-workflow' tab and click on the 'Send test' button.
+        </TextElement>
+      ),
+    },
+  ];
+}
```
Comment (djabarovgeorge) [inline apps/web/src/pages/get-started/tabs/Setup.tsx:339]: I wonder if we need this step here. I know we want to encourage users to trigger, but the trigger CTA is also located in the Studio flow. I think the comment in step 5 ('Run Novu Studio'), about the trigger might be enough.

I'm concerned that having six steps is too many. Based on my suggestions, we could reduce it to four, although I'm not sure if that's the best solution for user onboarding.

**Review by djabarovgeorge — APPROVED** [review]
Looks good, we would need to test it as much as possible in dev env!

**Pull request merged**